test/test_unit_check_logic.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+from unittest.mock import MagicMock, patch
+
+from conjur_api import Client
+from conjur.logic.check_logic import CheckLogic
+from conjur_api.errors.errors import HttpStatusError
+
+class CheckLogicTest(unittest.TestCase):
+
+    client = Client
+    check_logic = CheckLogic(client)
+
+    def test_check_logic_constructor(self):
+        mock_client = None
+        mock_check_logic = CheckLogic(mock_client)
+        self.assertEquals(mock_check_logic.client, mock_client)
+
+    @patch('conjur_api.Client.check_privilege')
+    def test_check_privilege_calls_client(self, mock_check):
+        self.check_logic.check('some_kind', 'some_resource_id', 'some_privilege', 'some_role_id')
+        mock_check.assert_called_once_with('some_kind', 'some_resource_id', 'some_privilege', 'some_role_id')
+
+    def test_check_privilege_missing_resource_parameter_raises_exception(self):
+        with self.assertRaises(TypeError):
+            self.check_logic.check('some_kind', 'some_privilege')
```

Comment:
If you use positional arguments here, the call to `check` doesn't know which parameter is missing. That makes this test identical to the one below `test_check_privilege_missing_privilege_parameter_raises_exception`

You can either remove one of these tests, or use named parameters (if supported) to make them functionally different. i.e.
`self.check_logic.check(kind: 'some_kind', privilege: 'some_privilege')`
and
`self.check_logic.check(kind: 'some_kind', resource: 'some_resource')`

Reply:
fixed. the prefixes kind:, privilege:, etc. wasn't supported so instead I broke apart two different cases,

a missing privilege or resource parameter test
a missing privilege and resource parameters test